### PR TITLE
endpoint: Validate endpoint properly using isDomainName and isValidIP

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

File: helpers.go [hunk 1]
```diff
@@ -34,6 +35,63 @@ var maxParts = int64(10000)
 // maxPartSize - maximum part size for a single multipart upload operation.
 var maxPartSize int64 = 1024 * 1024 * 1024 * 5
 
+// isDomainName validates input string if it conforms with
+// RFC 1035 and RFC 3696 for domain standards.
+func isDomainName(host string) bool {
```
Comment (abperiasamy) [inline helpers.go]: Use regex here.

File: helpers.go [hunk 2]
```diff
@@ -34,6 +35,34 @@ var maxParts = int64(10000)
...
+	// See RFC 1035, RFC 3696.
+	if len(host) == 0 {
+		// if string length '0' return false.
+		return false
+	}
+	if len(host) > 255 {
+		// if string larger than 255 characters return false.
+		return false
+	}
+	if host[(len(host)-1):] == "-" || host[:1] == "-" {
```
Comment (abperiasamy) [inline helpers.go]: This check is redundant. Regex below will catch this case as well.

File: helpers.go [hunk 3]
```diff
@@ -34,6 +35,34 @@ var maxParts = int64(10000)
 // maxPartSize - maximum part size for a single multipart upload operation.
 var maxPartSize int64 = 1024 * 1024 * 1024 * 5
 
+// isDomainName validates if input string is a valid domain name.
+func isDomainName(host string) bool {
+	// See RFC 1035, RFC 3696.
+	if len(host) == 0 {
```
Comment (abperiasamy) [inline helpers.go]: Regex can check for 1-255 range as well.

File: helpers.go [hunk 4]
```diff
@@ -34,6 +35,50 @@ var maxParts = int64(10000)
...
+	// host cannot start or end with "-"
+	if host[len(host)-1:] == "-" || host[:1] == "-" {
+		return false
+	}
+	// host cannot start or end with "_"
+	if host[len(host)-1:] == "_" || host[:1] == "_" {
+		return false
+	}
+	// host cannot start with a "."
+	if host[:1] == "." {
```
Comment (abperiasamy) [inline helpers.go]: Cannot end with "." as well.

File: helpers.go [hunk 5]
```diff
@@ -34,6 +35,50 @@ var maxParts = int64(10000)
...
+		return false
+	}
+	for _, r := range host {
```
Comment (abperiasamy) [inline helpers.go]: use strings.ContainsAny

**Comment by harshavardhana** [comment]
> Cannot end with "." as well.

Domains can end with "." - those are called absolute URLs

**Comment by harshavardhana** [comment]
> Domains can end with "." - those are called absolute domains.

http://www.dns-sd.org/trailingdotsindomainnames.html - for example a request to "https://www.example.com." is as valid as "https://www.example.com"

**Comment by harshavardhana** [comment]
> http://www.dns-sd.org/trailingdotsindomainnames.html - for example a request to "https://www.example.com." is as valid as "https://www.example.com"

You are right since this will be for our purposes, i do not see we need to be Browser like generic in validating domain names.

**Pull request merged**